File: 1014/main.go
package main

import "fmt"

func main() {
	fmt.Println(maxScoreSightseeingPair([]int{1,2,3,4}))
}

func maxScoreSightseeingPair(values []int) int {
	ans, mx := 0, values[0]+0
	for j := 1; j < len(values); j++ {
		ans = max(ans, mx + values[j]-j)
		mx = max(mx, values[j]+j)
	}
	return ans
}

File: 1014/main.rs
struct Solution {}

impl Solution {
    pub fn max_score_sightseeing_pair(values: Vec<i32>) -> i32 {
        let mut ans = 0;
        let mut mx = values[0] + 0;
        for j in 1..values.len() {
            ans = std::cmp::max(ans, mx + values[j as usize] - j as i32);
            mx = std::cmp::max(mx, values[j as usize] + j as i32);
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::max_score_sightseeing_pair(vec![1,2,3,4]));
}